internal/os.go

```diff
@@ -0,0 +1,66 @@
+package internal
```

Comment:
why not create a package for it? so that other internal packages can use it as well.